File: frontend/apps/web/src/apis/ai.ts
// ==========================================
// 1. 타입 정의 (Types)
// ==========================================

import api from './index';

// 요청 (Request)
export interface AIExhibitionRequest {
  prompt: string;
  ticketId: number;
}

// 내부 상세 데이터
export interface ExhibitionDesign {
  font: string;
  colorScheme: string;
  layoutType: string;
  frameStyle: string;
  background: string;
  backgroundImage: string;
}

export interface RecommendedMovie {
  movieId: number;
  curatorComment: string;
  posterUrl?: string;
}

// 응답 (Response)
export interface AIExhibitionResponse {
  resultJson: {
    title: string;
    curatorComment: string;
    design: ExhibitionDesign;
    movies: RecommendedMovie[];
    keywords: string[];
  };
}

// 큐레이션 영화 응답 타입
export interface CurateMoviesResponse {
  ticketId: number;
  movies: {
    movieId: number;
    title: string;
    posterUrl: string;
  }[];
}

// 영화 상세 요청/응답 타입
export interface MovieDetailRequest {
  movieId: number;
  theme: string;
}

export interface MovieDetailResponse {
  movieId: number;
  title: string;
  detail: string;
}

// ==========================================
// 2. API 호출 함수 (Service Logic)
// ==========================================

/**
 * 티켓 ID로 영화 목록 조회 (빠른 조회)
 * nginx 프록시 경로: /api/ai/curate-movies -> /api/v1/ai/curate-movies
 */
export const curateMovies = async (
  ticketId: number,
  limit: number = 5
): Promise<CurateMoviesResponse> => {
  const response = await api.post<CurateMoviesResponse>('/ai/curate-movies', {
    ticketId,
    limit,
  });
  return response.data;
};

/**
 * AI 전시회 생성
 * nginx 프록시 경로: /api/ai/generate -> /api/v1/ai/generate
 */
export const generateExhibition = async (
  prompt: string,
  ticketId: number
): Promise<AIExhibitionResponse> => {
  const response = await api.post<AIExhibitionResponse>('/ai/generate', {
    prompt,
    ticketId,
  });
  return response.data;
};

/**
 * 영화 상세 설명 생성 (포스터 클릭 시)
 * nginx 프록시 경로: /api/ai/movie-detail -> /api/v1/ai/movie-detail
 */
export const getMovieDetail = async (
  movieId: number,
  theme: string = '일반'
): Promise<MovieDetailResponse> => {
  const response = await api.post<MovieDetailResponse>('/ai/movie-detail', {
    movieId,
    theme,
  });
  return response.data;
};

File: backend/app/schemas/ai.py
"""
AI 관련 Pydantic 스키마
"""
from pydantic import BaseModel, ConfigDict
from typing import List


class ExhibitionDesign(BaseModel):
    """전시회 디자인"""
    font: str
    colorScheme: str
    layoutType: str
    frameStyle: str
    background: str
    backgroundImage: str

    model_config = ConfigDict(
        populate_by_name=True,
        json_schema_extra={
            "example": {
                "font": "Pretendard",
                "colorScheme": "dark",
                "layoutType": "grid",
                "frameStyle": "modern",
                "background": "#000000",
                "backgroundImage": "https://example.com/bg.jpg"
            }
        }
    )


class RecommendedMovie(BaseModel):
    """추천 영화"""
    movieId: int
    curatorComment: str

    model_config = ConfigDict(
        populate_by_name=True,
        json_schema_extra={
            "example": {
                "movieId": 101,
                "curatorComment": "감동적인 스토리와 뛰어난 연기가 돋보이는 작품입니다."
            }
        }
    )


class AIGenerateRequest(BaseModel):
    """AI 전시회 생성 요청"""
    prompt: str
    ticketId: int
    pinnedMovieIds: List[int] = []

    model_config = ConfigDict(
        json_schema_extra={
            "example": {
                "prompt": "감성적이고 잔잔한 영화를 추천해주세요",
                "ticketId": 3
            }
        }
    )


class AIGenerateResponse(BaseModel):
    """AI 전시회 생성 응답"""
    resultJson: dict

    model_config = ConfigDict(
        populate_by_name=True,
        json_schema_extra={
            "example": {
                "resultJson": {
                    "title": "잔잔한 감성 영화관",
                    "design": {
                        "font": "Pretendard",
                        "colorScheme": "soft",
                        "layoutType": "grid",
                        "frameStyle": "minimal",
                        "background": "#F5F5F5",
                        "backgroundImage": ""
                    },
                    "movies": [
                        {
                            "movieId": 101,
                            "curatorComment": "마음이 따뜻해지는 영화"
                        }
                    ],
                    "keywords": ["감성", "힐링", "잔잔함"]
                }
            }
        }
    )
